### PR TITLE
#4640 : The notification badge jumps around

### PR DESCRIPTION
**Pull request by ahmed-radhouane**
### Changes 
- change unread counter badge view in item_room to be in consistent place on the screen, to left of date/time.
 - change unread badge color state for dark and light theme to align with iOS.
 - add changelog file related to the issue.

Signed-off-by: Ahmed Radhouane Belkilani <arbelkilani@gmail.com>

### Screenshots 

|   RTL 	|   LTR	|   	
|---	|---	|
|   	|   	|   	
|    	|   	|   	

### Pull Request Checklist

- [x] Changes has been tested on an Android device or Android emulator with API 21
- [x] UI change has been tested on both light and dark themes
- [x] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [x] Pull request is based on the develop branch
- [x] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [x] Pull request includes screenshots or videos if containing UI changes
- [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [x] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: vector/src/main/res/layout/item_room.xml [hunk 1]
```diff
@@ -165,7 +165,6 @@
         android:visibility="gone"
         app:layout_constraintBottom_toBottomOf="@id/roomNameView"
         app:layout_constraintEnd_toStartOf="@id/roomLastEventTimeView"
-        app:layout_constraintStart_toEndOf="@id/roomDraftBadge"
```
Comment (ouchadam) [inline vector/src/main/res/layout/item_room.xml:168]: was this constraint the cause of the jumping?

Comment (ahmed-radhouane) [reply vector/src/main/res/layout/item_room.xml:168]: yes, it was end and start constrained with fixed size, which made the issue.

File: changelog.d/4640.bugfix [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# [[Issue #4640]](https://github.com/vector-im/element-android/issues/4640) The notification badge jumps around
```
Comment (ouchadam) [inline changelog.d/4640.bugfix:1]: for context, we're using https://github.com/twisted/towncrier to generate the release notes https://github.com/vector-im/element-android/releases and it automatically does the hyperlinking of issue in the file name for us :tada:

File: changelog.d/4640.bugfix [hunk 3]
```diff
@@ -0,0 +1,5 @@
...
+
+## Changes :
+- change unread counter badge view in item_room to be in consistent place on the screen, to left of date/time.
```
Comment (ouchadam) [inline changelog.d/4640.bugfix:4]: for the content of the entry we've been trying to keep them down to one liners, in this case we could have  

```
Fixing the room list notification badge jumping around
```

this content goes directly into the https://github.com/vector-im/element-android/blob/develop/CHANGES.md and release github release description

**Review by ouchadam — COMMENTED** [review]
small comment about the changelog entry, otherwise looks great, thanks! :100:

**Review by ouchadam — APPROVED** [review]

File: changelog.d/4640.bugfix (hunk 3 above)
Comment (ahmed-radhouane) [reply changelog.d/4640.bugfix]: It's updated. 
Thank you

File: library/ui-styles/src/main/res/values/colors.xml [hunk 4]
```diff
@@ -58,9 +58,9 @@
 
     <!-- Other colors, which are not in the palette -->
     <attr name="vctr_unread_room_badge" format="color" />
-    <color name="vctr_unread_room_badge_light">#FF61708B</color>
-    <color name="vctr_unread_room_badge_dark">#FF61708B</color>
-    <color name="vctr_unread_room_badge_black">#FF61708B</color>
+    <color name="vctr_unread_room_badge_light">@color/palette_gray_200</color>
+    <color name="vctr_unread_room_badge_dark">@color/palette_gray_250</color>
+    <color name="vctr_unread_room_badge_black">@color/palette_gray_250</color>
```
Comment (bmarty) [inline library/ui-styles/src/main/res/values/colors.xml:63]: Ok, so maybe we can remove `vctr_unread_room_badge` and the values and just use `vctr_content_secondary`.

**Pull request merged**